Implement `crates-admin` binary

Turbo87 opened the pull request:
This PR resolves https://github.com/rust-lang/crates.io/issues/1504 by adding a new `crates-admin` binary, that combines a lot of the previously separate binaries as subcommands.

I would recommend to review commit-by-commit :)

r? @jtgeibel

bors commented:
:umbrella: The latest upstream changes (presumably #2810) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#conflicts).

*Note that reviewers usually do not review pull requests until merge conflicts are resolved!* Once you resolve the conflicts, you should change the labels applied by bors to indicate that your PR is ready for review. Post this as a comment to change the labels:
```
@rustbot modify labels: +S-waiting-on-review -S-waiting-on-author
```

jtgeibel commented:
The `backfill-org-ids` bin was removed in #2810 and can be dropped here as well. Also we should be updating `bin_names` in `script/ci/prune-cache.sh` as we add/remove binaries. (This is mostly a hack to improve CI cache performance on Travis, and may be less necessary on GH Actions and with fewer binaries, but lets maintain it for now.)

For deployment purposes, `monitor` and `enqueue-job` are called from the scheduler on Heroku and it will be best to keep those around for a single deployment cycle. For the rest of the binaries included in this migration, I think it is fine to drop them in this PR.

Turbo87 commented:
@jtgeibel thanks for the review. I've removed `monitor` from the `crates-admin` tool, added it to the `prune-cache.sh` and removed the now obsolete binaries. :)

jtgeibel commented:
I pushed a commit dropping the about attribute on `populate` (it was leftover from `monitor`).

I'm fine with migrating `enqueue-job` and `monitor` as well. We could even do so in a single PR. Just deploy first commits (up to where the bins are removed), update the scheduler, then deploy the merged commit.

Merging this PR and deploying to staging.

@bors r+

bors commented:
:pushpin: Commit 9aa026c4892329719d4715dc48efc63ffbbc955d has been approved by `jtgeibel`

<!-- @bors r=jtgeibel 9aa026c4892329719d4715dc48efc63ffbbc955d -->
<!-- homu: {"type":"Approved","sha":"9aa026c4892329719d4715dc48efc63ffbbc955d","approver":"jtgeibel"} -->

bors commented:
:hourglass: Testing commit 9aa026c4892329719d4715dc48efc63ffbbc955d with merge 01cdee414b4cdca940ed71f0fee3c450751c5d2a...
<!-- homu: {"type":"BuildStarted","head_sha":"9aa026c4892329719d4715dc48efc63ffbbc955d","merge_sha":"01cdee414b4cdca940ed71f0fee3c450751c5d2a"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang/crates.io/runs/1393863548)
Approved by: jtgeibel
Pushing 01cdee414b4cdca940ed71f0fee3c450751c5d2a to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"jtgeibel","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang/crates.io/runs/1393863548"},"merge_sha":"01cdee414b4cdca940ed71f0fee3c450751c5d2a"} -->
